presto-main/src/main/java/com/facebook/presto/sql/analyzer/Scope.java

```diff
@@ -185,13 +194,28 @@ private static boolean isColumnReference(QualifiedName name, RelationType relati
         return Optional.empty();
     }
 
+    @Override
+    public String toString()
+    {
+        return toStringHelper(this)
+                .addValue(relationId)
+                .toString();
+    }
+
     public static final class Builder
     {
+        private RelationId relationId = RelationId.anonymous();
```

Comment:
Sure, but I prefer users to explicitly create scope without `relationId` (so they actually think what they are doing) instead of blindly creating scopes without `relationId` using `Scope.create()` method because it's quickest. This is fairly complex piece of code, therefore having a bit explicitness (instead of super short methods) is not a problem.

You could create `Scope.createAnonymous()` method to make people think twice before using it.